### PR TITLE
Develop upstream sync 240729

### PR DESCRIPTION
**Pull request by hsharsha**

### CODE REVIEW

**Comment by i-chaochen** [comment]
```
ERROR: Skipping '//tensorflow/python/integration_testing/...': no targets found beneath 'tensorflow/python/integration_testing'
ERROR: no targets found beneath 'tensorflow/python/integration_testing'
```


seems we need to update [bazelrc](https://github.com/ROCm/tensorflow-upstream/blob/develop-upstream/tensorflow/tools/tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc#L32-L81)

File: third_party/xla/third_party/tsl/tsl/platform/default/dso_loader.cc [hunk 1]
```diff
@@ -195,19 +195,9 @@ absl::StatusOr<void*> GetHipblasltDsoHandle() {
...
 
-<<<<<<< HEAD
-#if (TF_ROCM_VERSION >= 60000)
-#define HIP_SO_VERSION "6"
-#else
-#define HIP_SO_VERSION "5"
-#endif
-
-absl::StatusOr<void*> GetHipDsoHandle() { return GetDsoHandle("amdhip64", HIP_SO_VERSION); }
-=======
```
Comment (mmakevic-amd) [inline third_party/xla/third_party/tsl/tsl/platform/default/dso_loader.cc]: Do we have a proper fix for this now? If not we will have issues with trying to load two versions of the same library (https://github.com/ROCm/tensorflow-upstream/commit/cf09f6adea48688d1775cf3a272377239a4c5f40)

Comment (i-chaochen) [reply third_party/xla/third_party/tsl/tsl/platform/default/dso_loader.cc]: are you referring this https://github.com/openxla/xla/pull/13879 and https://github.com/openxla/xla/commit/69437a383f65961279f01fcd497ad55fd242bd0a ?

Comment (mmakevic-amd) [reply third_party/xla/third_party/tsl/tsl/platform/default/dso_loader.cc]: Yes, thank you! Not sure how I missed it during the last sync 😅

File: third_party/xla/xla/tests/BUILD [hunk 2]
```diff
@@ -1102,11 +1099,8 @@ xla_test(
     shard_count = 20,
     tags = [
         "optonly",
-<<<<<<< HEAD
         "no_rocm", #TODO(rocm): TEMP, sync 24-06-24
```
Comment (mmakevic-amd) [inline third_party/xla/xla/tests/BUILD]: Can we re-enable dot tests now? I see they were passing on CI on 30/07 https://github.com/ROCm/frameworks-internal/issues/8124#issue-2321996330

**Comment by hsharsha** [comment]
retest gpu-pycpp please

**Comment by hsharsha** [comment]
> ```
> ERROR: Skipping '//tensorflow/python/integration_testing/...': no targets found beneath 'tensorflow/python/integration_testing'
> ERROR: no targets found beneath 'tensorflow/python/integration_testing'
> ```
> 
> seems we need to update [bazelrc](https://github.com/ROCm/tensorflow-upstream/blob/develop-upstream/tensorflow/tools/tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc#L32-L81)

@i-chaochen  I updated [devel.usertools/gpu.bazelrc](https://github.com/ROCm/tensorflow-upstream/pull/2599/commits/efff3d8805d174ac90e01dc02ef2bde85ba940c3) but still seeing same failure

**Comment by i-chaochen** [comment]
Yes, I think it's because the docker container `rocm/tensorflow-build:latest-focal-python3.9-rocm6.1.0` doesn't update...we need to manually rebuild it here http://ml-ci.amd.com:21096/job/auto-docker-builds/job/tensorflow-build/job/tensorflow-build-develop-upstream-ub20/    cc @jayfurmanek

**Comment by hsharsha** [comment]
retest gpu-pycpp please

**Comment by hsharsha** [comment]
retest gpu-pycpp please

**Review by i-chaochen — COMMENTED** [review]
so, is this a previous or newer llvm commit compared to this weekly-sync's llvm commit?

**Comment by hsharsha** [comment]
> so, is this a previous or newer llvm commit compared to this weekly-sync's llvm commit?

It is taken from https://github.com/tensorflow/tensorflow/pull/71729  which is 2-3 weeks older than the weekly-sync's llvm commit

**Comment by i-chaochen** [comment]
> > so, is this a previous or newer llvm commit compared to this weekly-sync's llvm commit?
> 
> It is taken from [tensorflow#71729](https://github.com/tensorflow/tensorflow/pull/71729) which is 2-3 weeks older than the weekly-sync's llvm commit

We should try the newer llvm commit to address the error instead of older one (workable) in weekly-sync.

**Comment by hsharsha** [comment]
> > > so, is this a previous or newer llvm commit compared to this weekly-sync's llvm commit?
> > 
> > 
> > It is taken from [tensorflow#71729](https://github.com/tensorflow/tensorflow/pull/71729) which is 2-3 weeks older than the weekly-sync's llvm commit
> 
> We should try the newer llvm commit to address the error instead of older one (workable) in weekly-sync.

None of the new ones or even a bit older ones do not work with jit=False. They emit `builtin.unrealized_conversion_cast` which cannot be handled by LLVM.  Only this LLVM commit seems to work.

**Comment by hsharsha** [comment]
retest gpu-pycpp please

**Comment by hsharsha** [comment]
retest cpu-pycpp please

**Comment by hsharsha** [comment]
retest cpu-pycpp please

**Comment by hsharsha** [comment]
retest gpu-pycpp please

**Comment by hsharsha** [comment]
retest cpu-pycpp please

**Comment by hsharsha** [comment]
retest cpu-pycpp please

**Review by mmakevic-amd — APPROVED** [review]

**Pull request merged**